readthedocs/projects/tasks/builds.py

```diff
@@ -554,6 +555,16 @@ def get_valid_artifact_types(self):
                 version=self.data.version.slug,
                 type_=artifact_type,
             )
+
+            if artifact_type == "html":
+                index_html_filepath = os.path.join(artifact_directory, "index.html")
+                if not os.path.exists(index_html_filepath):
+                    log.info(
+                        "Failing the build. "
+                        "HTML output does not contain an 'index.html' at its root directory.",
+                    )
+                    raise BuildUserError(BuildUserError.BUILD_OUTPUT_HTML_NO_INDEX_FILE)
```

Comment:
Is there a way to easily notify users and not fail the build, to start? I feel like we have a pretty good pattern there, if it's easy, and I would like to start doing that any time we remove something that will hard fail a user, to at least give them a heads up with a short duration (eg. 2-4 weeks).

Reply:
If we are worried about how many people this will affect, I'd suggest to

- comment this line here 
- add a `log.warning` to collect projects on NR
- wait some weeks
- send those users an email (following the script pattern) and give them 1 month
- uncomment the line again and deploy the changes

Reply:
That seems like a good step 👍